py3status/modules/updates.py

```diff
@@ -0,0 +1,215 @@
+
+35pkgs
+[{'full_text': 'Zypper '}, {'full_text': '34', 'color': '#ffa500'}]
+
+45pkgs
+[{'full_text': 'Xbps '}, {'full_text': '45', 'color': '#FF0000'}]
+
+no_updates
+{'full_text': 'No Updates'}
+"""
+
+STRING_INVALID = 'invalid manager'
+
+
+class Update:
+    def __init__(self, parent, name, command):
+        self.parent = parent
+        self.name = name
+        self.command = command
+
+    def get_output(self):
+        try:
+            return self.parent.py3.command_output(self.command)
+        except self.parent.py3.CommandError:
+            return None
+
+    def count_updates(self, output):
+        return len(output.splitlines())
+
+    def get_updates(self):
+        output = self.get_output()
+        if output is None:
+            return {self.name: output}
+        return {self.name: self.count_updates(output)}
+
+
+class Apt(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Apk(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Eopkg(Update):
+    def get_output(self):
+        output = self.parent.py3.command_output(self.command)
+        if "No packages to upgrade." in output:
+            return ""
+        return output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    thresholds = [
+        (0, 'darkgray'), (10, 'degraded'), (20, 'orange'), (30, 'bad')]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        managers = update_managers = [
+            ("Pacman", "checkupdates"),
+            ("Auracle", "auracle sync --color=never"),
+            ("Yay", "yay --query --upgrades --aur"),
+            ("Apk", "apk version -l '<'"),
+            ("Apt", "apt list --upgradeable"),
+            ("Eopkg", "eopkg list-upgrades"),
+            ("Pkg", "pkg upgrade --dry-run --quiet"),
+            ("Xbps", "xbps-install --update --dry-run"),
+            ("Zypper", "zypper list-updates"),
+        ]
+
+        managed = getattr(self, 'managers', None)
+        if managed:
+            new_managers = []
+            for custom in managed:
+                if isinstance(custom, tuple):
+                    if len(custom) != 2 or custom[0].lower() == 'update':
+                        raise Exception(STRING_INVALID)
+                    new_managers.append(custom)
+                else:
+                    name = custom.lower()
+                    if name == 'update':
+                        raise Exception(STRING_INVALID)
```

Comment:
same comment about `managers` not containing `update`

Reply:
You would think so, but #1543 tells me that users will try `update`. I have to add code there eventually to reject `mail` and here, to reject `update` because I can see users wanting custom `Update {update}` in `managers`

Reply:
Got it, update is reserved keyword and you are guarding against it here 👍